dynatrace/api/v2/settings/objects/permissions/settings_client_test.go

```diff
@@ -0,0 +1,327 @@
+				require.Equal(t, request.URL.Query().Get("fields"), "schemaId,ownerBasedAccessControl")
+				require.Equal(t, http.MethodGet, request.Method)
+				require.Equal(t, "/platform/classic/environment-api/v2/settings/schemas", request.URL.Path)
+
+				_, err := writer.Write([]byte(`{"items":[{"schemaId":"schemaID1","ownerBasedAccessControl":true},{"schemaId":"schemaID2","ownerBasedAccessControl":false}]}`))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+
+			url, _ := url.Parse(server.URL)
+			client := permissions.NewSettingsClient(rest.NewClient(url, server.Client()))
+
+			schemaIDs, err := client.GetSchemaIDsWithOwnerBasedAccessControl(t.Context())
+
+			assert.NoError(t, err)
+			assert.Len(t, schemaIDs, 1)
+			assert.Contains(t, schemaIDs, "schemaID1")
+		})
+
+		t.Run("returns error if response is not 200", func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				writer.WriteHeader(http.StatusInternalServerError)
+			}))
+			defer server.Close()
+
+			url, _ := url.Parse(server.URL)
+			client := permissions.NewSettingsClient(rest.NewClient(url, server.Client()))
+
+			schemaIDs, err := client.GetSchemaIDsWithOwnerBasedAccessControl(t.Context())
+
+			assert.Error(t, err)
+			assert.Nil(t, schemaIDs)
+		})
+
+		t.Run("returns error if response is not JSON", func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				_, err := writer.Write([]byte(`{items:[]}`))
+				require.NoError(t, err)
+			}))
+			defer server.Close()
+
+			url, _ := url.Parse(server.URL)
+			client := permissions.NewSettingsClient(rest.NewClient(url, server.Client()))
+
+			schemaIDs, err := client.GetSchemaIDsWithOwnerBasedAccessControl(t.Context())
+
+			assert.Error(t, err)
+			assert.Nil(t, schemaIDs)
+		})
+
+		t.Run("returns error if connection fails", func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {}))
+
+			url, _ := url.Parse(server.URL)
+			client := permissions.NewSettingsClient(rest.NewClient(url, server.Client()))
+			server.Close()
+
+			schemaIDs, err := client.GetSchemaIDsWithOwnerBasedAccessControl(t.Context())
+
+			assert.Error(t, err)
+			assert.Nil(t, schemaIDs)
+		})
+	})
+
+	t.Run("ListObjectsIDsOfSchema", func(t *testing.T) {
+
+		t.Run("retries without admin access if first request fails with forbidden", func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				require.Equal(t, request.URL.Query().Get("schemaIds"), "schemaID1")
+				require.Equal(t, request.URL.Query().Get("fields"), "objectId")
+				require.Empty(t, request.URL.Query().Get("nextPageKey"))
+				require.Equal(t, http.MethodGet, request.Method)
+				require.Equal(t, "/platform/classic/environment-api/v2/settings/objects", request.URL.Path)
```

Comment:
[nit]: Should we introduce a counter here? At the moment, the test would still succeed if the `adminAccess` is not set at all.
Also, instead of asserting the path and method, we could also just use a server mux (also for other test cases), or is there a limitation with the query params?